File: python/meteroid/models/create_product_request.py
# this file is @generated
import dataclasses
import typing as t

from ..serialization import BaseModel
from .product_family_id import ProductFamilyId
from .product_fee_structure import ProductFeeStructure


@dataclasses.dataclass
class CreateProductRequest(BaseModel):
    fee_structure: ProductFeeStructure

    name: str

    product_family_id: ProductFamilyId

    catalog: t.Optional[bool] = None

    description: t.Optional[str] = None

File: go/create_product_request.go
// this file is @generated
package meteroid

type CreateProductRequest struct {
	Catalog *bool `json:"catalog,omitempty"`

	Description *string `json:"description,omitempty"`

	FeeStructure ProductFeeStructure `json:"fee_structure"`

	Name string `json:"name"`

	ProductFamilyId ProductFamilyId `json:"product_family_id"`
}
